proof/invariant-abstract/Invariants_AI.thy

```diff
@@ -1276,10 +1276,8 @@ definition
   "cur_tcb s \<equiv> tcb_at (cur_thread s) s"
 
 definition
-  "cur_sc_tcb s \<equiv>
-     sc_tcb_sc_at
-       (\<lambda>t. scheduler_action s = resume_cur_thread \<longrightarrow> t = Some (cur_thread s))
-       (cur_sc s) s"
+  "cur_sc_tcb s
+    \<equiv> (scheduler_action s = resume_cur_thread) \<longrightarrow> sc_tcb_sc_at (\<lambda>t. t = Some (cur_thread s)) (cur_sc s) s"
```

Comment:
```suggestion
  "cur_sc_tcb s
    \<equiv> scheduler_action s = resume_cur_thread \<longrightarrow> sc_tcb_sc_at (\<lambda>t. t = Some (cur_thread s)) (cur_sc s) s"
```

Could potentially also write
```
    \<equiv> scheduler_action s = resume_cur_thread \<longrightarrow> sc_tcb_sc_at ((=) Some (cur_thread s)) (cur_sc s) s"
```
but that will turn the equality around, i.e make it into `Some .. = t`. Potentially less nice in the proofs.

Reply:
I originally had it like this, but then something wasn't working and I became paranoid about precedence. But I've made the change you've suggested

Reply:
If you are unsure about precedence, you can use the `term` command with more parentheses to let Isabelle echo back to you what the "standard" parentheses are. I.e. if Isabelle leaves them out in the output and you have had them in the input, they are safe to leave out.

Reply:
Oooh that's a good idea, thanks